src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,63 @@
+package duke.task;
+
+/**
+ * Task class, for inheritance of various tasks.
+ */
+public class Task {
+    private String description;
+    private boolean isDone = false;
+
+    /**
+     * Constructs a Task object.
+     *
+     * @param description Description of task.
+     */
+    public Task(String description) {
+        this.description = description;
+    }
+
+    /**
+     * Returns status icon of the task.
+     *
+     * @return the tick icon [✓] if task is done, or cross icon [✗] otherwise.
+     */
+    private String getStatusIcon() {
+        //CHECKSTYLE.OFF: AvoidEscapedUnicodeCharactersCheck
+        if (this.isDone) {
+            return "[\u2713] ";
+        } else {
+            return "[\u2718] ";
+        }
+        //CHECKSTYLE.ON: AvoidEscapedUnicodeCharactersCheck
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Extracts description of the task.
+     *
+     * @return description of the task.
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Gives a string representation of the task for storage.
+     *
+     * @return a string representation of the task for storage.
+     */
+    public String toDataString() {
+        return (this.isDone ? 1 : 0) + " | " + this.description;
```

Comment:
I have a minor comment here, I am sure that this code here works perfectly, but this explanation might not be as clear and straightforward than if else form. I feel this way, do you agree?